app/src/main/java/com/teamagam/gimelgimel/app/map/view/DrawActionActivity.java

```diff
@@ -76,8 +83,10 @@ private BaseDrawActionFragment setupActionFragment(String action) {
   private BaseDrawActionFragment getActionFragment(String action) {
     if (SEND_QUAD_ACTION.equalsIgnoreCase(action)) {
       return new SendQuadrilateralActionFragment();
-    } else {
+    } else if (MEASURE_DISTANCE_ACTION.equalsIgnoreCase(action)) {
       return new MeasureActionFragment();
+    } else {
```

Comment:
I think u should check the string to be SEND_GEOMETRY_ACTION even though it can't be invalid theoretically.
If I for example add a new action and I setupActionFragment(MY_NEW_ACTION) I will get the SendGeoFragment and will be surprised. Instead you can log an error in the "else" scope and I know where to go...